docs/WindowsBuild.md

```diff
@@ -0,0 +1,177 @@
+# Building Swift on Windows
+
+This document describes how to build Swift for Windows natively. For more
+information about Swift on Windows, see [the Windows doc](./Windows.md) for
+more context of what's possible on Windows.
+
+There are two supported ways to build Swift on Windows, they are
+
+1. Using [`clang-cl`](https://clang.llvm.org/docs/UsersManual.html#clang-cl)
+1. Using Microsoft Visual Studio Compiler (MVSC)
+
+Using `clang-cl` is recommended over MVSC for building Swift on Windows.
+Although it is possible to build the compiler and the standard library with
+MVSC, and use those built products to compile a Swift program, it won't be
+possible to run the binary without seperately obtaining the Swift runtime. On
+the other hand, `clang-cl` is able to build the runtime, which makes it
+possible to build and run all the components required for Swift natively on
+Windows.
+
+## clang-cl
+- Windows doesn't currently have a build script. You'll need to run commands
+  manually to build Swift on Windows.
+- Windows support for Swift is a work in progress and may not work on your
+  system, but it has been tested.
+- Using the latest Visual Studio version is recommended. Swift may fail to
+  build with older C++ compilers.
+
+### 1. Install dependencies
+- Make sure to add Python, CMake and Ninja to your `Path` environment variable
+1. Latest version (2.7.12 tested) of [Python
+2](https://www.python.org/downloads/)
+2. Latest version (3.7.0-rc3 tested) of [CMake](https://cmake.org/download/)
+3. Latest version (1.7.1 tested) of
+[Ninja](https://github.com/ninja-build/ninja/releases/latest)
+4. Latest version (2015 Update 3 tested) of [Visual
+Studio](https://www.visualstudio.com/downloads/)
+- Make sure to include `Programming Languages|Visual C++` and `Windows and Web
+  Development|Universal Windows App Development|Windows SDK` in your
+  installation.
+- Windows SDK 10.0.10240 was tested. Some later versions (e.g. 10.0.14393) are
```

Comment:
Not sure what compiler-rt has to do with this.  compiler-rt definitely should be okay with 14393.